pkg/apis/core/v1/defaults.go

```diff
@@ -429,3 +435,55 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+// defaultPodResources applies default values for pod-level requests, when pod-level
+// limits are set, in following scenarios:
+// 1. When at least one container has requests set: The pod-level requests become
+// equal to the effective requests of all containers in the pod.
+// 2. When no containers have requests set: The pod-level requests become equal to
+// pod-level limits.
+// This defaulting behavior ensures consistent resource accounting at the pod-level
+// while mainting compatability with the container-level specifications, as detailed
+// in KEP-2837: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2837-pod-level-resource-spec/README.md#proposed-validation--defaulting-rules
+func defaultPodResources(obj *v1.Pod) {
+	if obj.Spec.Resources.Limits == nil || len(obj.Spec.Resources.Limits) > 0 {
+		return
+	}
+
+	var reqs v1.ResourceList
+	reqs = obj.Spec.Resources.Requests
+	if obj.Spec.Resources.Requests == nil {
+		reqs = make(v1.ResourceList)
+	}
+
+	podReqsFromContainers := resourcehelper.PodRequests(obj, resourcehelper.PodResourcesOptions{
+		InPlacePodVerticalScalingEnabled: utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling),
+		PodLevelResourcesEnabled:         utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources),
+		UseContainerLevelResources:       true,
```

Comment:
I'm confused, why do we have this field? will it be an additional feature gate or something?